libraries/ms-common/inc/ms_test_helper_datagram.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+/* Test helpers for datagrams
+ */
+
+#include "bootloader_can.h"
+#include "can_datagram.h"
+#include "ms_test_helpers.h"
+#include "status.h"
+
+#define DATAGRAM_PROCESS_ALL(e)                                 \
+  while (can_datagram_get_status() == DATAGRAM_STATUS_ACTIVE) { \
+    MS_TEST_HELPER_AWAIT_EVENT(e);                              \
+    can_datagram_process_event(&e);                             \
+    can_process_event(&e);                                      \
+  }
```

Comment:
Could you wrap this in either `do { ... } while (0)` or `({ })` (GNU extension)? That way things don't get weird with scope and semicolons. Also consider adding a documentation comment above it.